File: 4thSemester/Grafos/ImplementacaoCpp/vertice.cpp
#include <string>
#include <iostream>
#include "./header files/vertice.hpp"

Vertice::Vertice(std::string rotulo){
    this->rotulo = rotulo;
    this->grau_incidencia = this->grau_saida = 0;
}

std::string Vertice::toString(){
    std::string str = "\nRotulo: " + this->rotulo;
    str += "\nGrau de incidecia" + std::to_string(this->grau_incidencia); 
    str += "\nGrau de saida" + std::to_string(this->grau_saida); 
    return str;
}

void Vertice::add_grau_incidencia(){
    this->grau_incidencia++;
}

void Vertice::add_grau_saida(){
    this->grau_saida++;
}

//incrementa ambos os graus (exclusivo para grafos nao direcionados)
void Vertice::add_grau(){
    this->grau_incidencia += 1;
    this->grau_saida = grau_incidencia;
}

//metodo criado apenas para facilitar a legibilidade do código
int Vertice::get_grau(){
    return this->grau_incidencia;
}

int Vertice::get_grau_saida(){
    return this->grau_saida;
}

int Vertice::get_grau_incidencia(){
    return this->grau_incidencia;
}

std::string Vertice::get_rotulo(){
    return this->rotulo;
}

File: 4thSemester/Grafos/implementacaoPython/vertice.py
class Vertice: 
    def __init__(self, rotulo):
        self._rotulo = rotulo
        self._grau_incidencia = self._grau_saida = self._grau = 0

    @property
    def rotulo(self):
        return self._rotulo
    
    @property
    def grau_incidencia(self):
        return self._grau_incidencia

    @property
    def grau_saida(self):
        return self._grau_saida
    
    @property
    def grau(self):
        return self._grau

    def update_grau_incidencia(self):
        self._grau_incidencia += 1
    
    def update_grau_saida(self):
        self._grau_saida += 1
    
    def update_grau(self):
        self._grau += 1
        self._grau_saida = self._grau_incidencia = self._grau
            
    def to_string(self):
        result = f"Rotulo: {self.rotulo}\n"
        result += f" - Grau entrada: {self._grau_incidencia}\n"
        result += f" - Grau saida: {self._grau_saida}"
        return result;
